### PR TITLE
feat(scouts): cross-fleet scout findings page

### PR DESCRIPTION
**Pull request by andrewm4894**
this is bringing code app scouts pages in line with the could inbox scouts pages where you have a place to see all recent scout signals

## Problem

The Code app's scouts feature surfaces findings (emissions) only inline on a single scout's detail page and inside individual inbox reports. There's no way to see every finding the whole troop emitted recently in one place — which PostHog Cloud's inbox already offers via its `FindingsPanel` at `/inbox/scouts/findings`. This adds the equivalent cross-fleet findings surface to the Code app.

## Changes

image

Adds a cross-fleet **Scout findings** page at `/code/agents/scouts/findings`, mirroring Cloud's `FindingsPanel` but built in the Code app's idioms (React Query hooks + TanStack file routes + Radix/Tailwind + pure `@posthog/core` helpers — not Kea).

- **`packages/core/src/scouts/scoutFindings.ts`** — host-agnostic aggregation logic (the pure counterpart of Cloud's `findingsLogic` selectors): narrow the runs window to recent emitted runs, join each emission to its run + linked inbox report, derive the scout filter list, and search/filter/sort the flat row list. Fully unit-tested.
- **`useScoutFindings.ts`** — fans out one emissions + one report-link query per emitted run via `useQueries` and flattens through the core helper. (The per-scout `ScoutSignalsSection` renders a child component per run, which can't produce one sortable list.) Distinguishes partial vs. total fetch failure so the page can warn when the list is incomplete.
- **`ScoutFindingsView.tsx`** — the page: header, search box, scout/severity/sort selectors, and a list of `ScoutEmissionCard`s with discuss/share/task-run actions and the linked-report chip.
- **`FleetFindingsCallout.tsx`** — entry-point stat card on the fleet section (beside the existing scratchpad callout), reading the cheap `emitted_count` sum; renders nothing until the troop has emitted something.
- **`ScoutEmissionCard.tsx`** — new optional `showScout` prop so cross-fleet cards title with the emitting scout instead of the generic "Finding".
- Route registered and `routeTree.gen.ts` regenerated.

Read-only surface; acting on a finding still happens in its inbox report. No backend, host, or DI changes.

## How did you test this?

- `pnpm --filter @posthog/core typecheck` and `pnpm --filter @posthog/ui typecheck` — clean
- `pnpm --filter @posthog/core test scoutFindings` — 19 unit tests pass (aggregation, filtering, sorting, edge cases)
- `biome check` clean on all touched files; `noRestrictedImports` clean on core
- Pre-commit hook re-ran Biome + full `pnpm typecheck` — passed
- Not yet exercised in the running app (handed to the author to dogfood locally)

## Automatic notifications

- [ ] Publish to changelog?
- [ ] Alert Sales and Marketing teams?

### CODE REVIEW

File: packages/core/src/scouts/scoutFindings.ts [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+ */
+export function mostRecentEmittedRuns(
+  runs: ScoutRun[],
+  cap: number = FINDINGS_MAX_EMITTED_RUNS,
+): ScoutRun[] {
+  return runs
+    .filter((run) => (run.emitted_count ?? 0) > 0)
+    .slice()
+    .sort(byRunStartedDesc)
+    .slice(0, cap);
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/core/src/scouts/scoutFindings.ts:70]: Good catch — `.filter()` already returns a fresh array, so that intermediate `.slice()` was dead. Removed in e7369ba7c.

File: packages/core/src/scouts/scoutFindings.ts [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      const haystack =
+        `${row.emission.description ?? ""} ${prettifyScoutSkillName(row.run.skill_name)}`.toLowerCase();
+      if (!haystack.includes(needle)) {
+        return false;
+      }
+    }
+    return true;
+  });
+
+  return filtered.slice().sort((a, b) => {
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/core/src/scouts/scoutFindings.ts]: Agreed, same redundant copy. Removed in e7369ba7c.

File: packages/ui/src/features/scouts/hooks/useScoutFindings.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      },
+      enabled: !!client && !!projectId,
+      staleTime: 60_000,
+      meta: AUTH_SCOPED_QUERY_META,
+    })),
+  });
+
+  const emissions = useMemo(
+    () => emissionsResults.flatMap((result) => result.data ?? []),
+    [emissionsResults],
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/ui/src/features/scouts/hooks/useScoutFindings.ts:108]: Right that it was dead as written — but the intent was to *surface* it, not drop it. Fixed in e7369ba7c by consuming it: the hook now exposes `windowTruncated` (pagination stopped early **or** the emitted-run set exceeded the fan-out cap), and the page renders an amber "showing the most recent runs only…" banner when it's true. See also the Codex thread on the same line.

File: packages/core/src/scouts/scoutFindings.test.ts [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+        (r) => r.emission.id,
+      ),
+    ).toEqual(["b", "c", "a"]);
+  });
+
+  it("filters by scout", () => {
+    expect(
+      filterAndSortFindings(rows, {
+        ...base,
+        scoutFilter: "signals-scout-web-analytics",
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/core/src/scouts/scoutFindings.test.ts]: Done in e7369ba7c — the four sort cases are now a single `it.each` over `[sortKey, expectedOrder]`, and the two filter cases collapsed into an `it.each` too. Left the search test standalone since it asserts two distinct queries with different intent.

File: packages/ui/src/features/scouts/hooks/useScoutFindings.ts [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  const projectId = useAuthStateValue((state) => state.currentProjectId);
+  const runsQuery = useScoutRuns();
+
+  const emittedRuns = useMemo(
+    () => mostRecentEmittedRuns(runsQuery.data?.runs ?? []),
+    [runsQuery.data],
+  );
+
+  const emissionsResults = useQueries({
+    queries: emittedRuns.map((run) => ({
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/ui/src/features/scouts/hooks/useScoutFindings.ts:67]: The portable business logic *is* in `@posthog/core` — `buildFindingRows` / `filterAndSortFindings` / `mostRecentEmittedRuns`, with unit tests. What's left in the hook is React Query wiring: a `useQueries` fan-out that can't move to core (core forbids React / trpc / react-query imports per the import rules).

There's also no `ScoutsService` to move it into — every scout surface fetches via React Query hooks in `@posthog/ui` (`useScoutRuns`, `useScoutRunEmissions`, `useScoutEmissionReports`), and the per-scout `ScoutSignalsSection` does the same multi-run fan-out. Putting this one hook behind a service would diverge from the established scouts pattern without making anything more portable. Happy to revisit if/when we introduce a core scouts-service layer across the board.

File: packages/ui/src/features/scouts/hooks/useScoutFindings.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+      queryKey: scoutQueryKeys.emissions(projectId, run.run_id),
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/ui/src/features/scouts/hooks/useScoutFindings.ts]: Fixed in e7369ba7c — the emissions query key now includes `run.emitted_count`, so an in-progress run that emits more produces a new key and refetches, while completed runs keep a stable count (and stable key, no refetch). Kept the shared `scoutQueryKeys.emissions(...)` as the key prefix so prefix-targeted invalidation still reaches it.

File: packages/ui/src/features/scouts/components/ScoutFindingsView.tsx [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+                  timestamp={latest}
+                  className="text-[12px] text-gray-10"
+                />
+              </>
+            ) : null}
+          </Flex>
+        ) : null}
+        <Text className="text-[12px] text-gray-9">
+          Covers findings from the most recent {SCOUT_RUNS_WINDOW_SPAN} of troop
+          runs. Older findings live on in the inbox reports they produced.
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/ui/src/features/scouts/components/ScoutFindingsView.tsx:133]: Addressed in e7369ba7c. The hook now returns `windowTruncated` (pagination stopped early, or more emitted runs than the `FINDINGS_MAX_EMITTED_RUNS` fan-out cap) alongside `incompleteRuns`; the page shows an amber banner when either is set, so the header counts no longer imply completeness. Left the "most recent 3 days" copy as the nominal window, with the banner carrying the caveat when the set is actually clipped.

File: packages/ui/src/features/scouts/components/ScoutFindingsView.tsx [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+        >
+          Retry
+        </button>
+      </Flex>
+    );
+  }
+
+  if (rows.length === 0) {
+    return (
+      <Box className="rounded-(--radius-2) border border-(--gray-6) border-dashed bg-gray-1 px-4 py-8 text-center text-[12.5px] text-gray-11">
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/ui/src/features/scouts/components/ScoutFindingsView.tsx:295]: Kept deliberately in parity with the sibling `ScratchpadView` (merged in #2973), which hand-rolls the identical inline error/empty boxes for the same scout surface. These are inline states inside a populated, filterable list (search + filters sit above), not a full-screen placeholder — and "no findings match your filters" doesn't map cleanly onto quill `Empty`'s icon → title → CTA shape. Converting only this page would split the two scout surfaces stylistically.

I'd rather migrate `ScratchpadView` **and** this page to quill `Empty` together in a follow-up so they stay consistent — I'll file that.

File: packages/ui/src/features/scouts/hooks/useScoutFindings.ts [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+        return client.listScoutEmissionReports(projectId, run.run_id);
+      },
+      enabled: !!client && !!projectId,
+      staleTime: 60_000,
+      meta: AUTH_SCOPED_QUERY_META,
+    })),
+  });
+
+  const emissions = useMemo(
+    () => emissionsResults.flatMap((result) => result.data ?? []),
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/ui/src/features/scouts/hooks/useScoutFindings.ts:107]: Fixed in e7369ba7c — `incompleteRuns` now counts runs whose emissions fetch *errored* **or** resolved empty despite `emitted_count > 0`, and the page warns when it's non-zero (while still reporting the all-failed case as a hard error state). That mirrors the per-scout section's "no signal details available for this run" handling rather than silently dropping the run.

**Comment by andrewm4894** [comment]
Re the **React Doctor** `jsx-no-jsx-as-prop` warning on `ScoutFindingsView.tsx:328` (`footerEnd={<ScoutTaskRunLink … />}`): intentional and unchanged. `ScoutEmissionCard` exposes `actions` and `footerEnd` as `ReactNode` render slots, and the existing `ScoutSignalsSection` passes JSX to them the same way — this keeps the findings page consistent with that card's API. Inlining the link or restructuring to avoid the prop would only diverge from the established usage.

File: packages/ui/src/features/scouts/components/FleetFindingsCallout.tsx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+ * cross-fleet findings browse/filter surface. Reads the cheap `emitted_count`
+ * sum off the already-polled runs window (no per-run emissions fan-out, that's
+ * the page's job). Renders nothing until there's at least one finding, so a
+ * fresh project isn't nudged toward an empty page.
+ */
+export function FleetFindingsCallout() {
+  const { data: runsWindow } = useScoutRuns();
+
+  const emittedRuns = useMemo(
+    () => mostRecentEmittedRuns(runsWindow?.runs ?? []),
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/ui/src/features/scouts/components/FleetFindingsCallout.tsx]: Good catch — fixed in 2f5595c88. The callout was reusing the capped `mostRecentEmittedRuns` helper; it now filters emitted runs **uncapped** and sums `emitted_count` over the whole window (and derives "latest" by max timestamp rather than the capped head). The cap stays where it belongs — the findings page's per-run fan-out.

File: packages/ui/src/features/scouts/hooks/useScoutFindings.ts [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+  // the page can warn rather than imply completeness. Only meaningful when it
+  // isn't a total failure (that surfaces as `isError`).
+  const incompleteRuns = allEmissionsFailed
+    ? 0
+    : emittedRuns.reduce((count, run, index) => {
+        const result = emissionsResults[index];
+        if (!result) return count;
+        if (result.isError) return count + 1;
+        const empty = (result.data?.length ?? 0) === 0;
+        if (result.isSuccess && empty && (run.emitted_count ?? 0) > 0) {
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/ui/src/features/scouts/hooks/useScoutFindings.ts]: Fixed in 2f5595c88 — `incompleteRuns` now flags a run when `loaded < (emitted_count ?? 0)`, so a partial response (promised 5, returned 3) counts as incomplete, not just the fully-empty case. The empty case is subsumed by the same comparison.

File: packages/ui/src/features/scouts/hooks/useScoutFindings.ts [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+      },
+      enabled: !!client && !!projectId,
+      staleTime: 60_000,
+      meta: AUTH_SCOPED_QUERY_META,
+    })),
+  });
+
+  const reportResults = useQueries({
+    queries: emittedRuns.map((run) => ({
+      queryKey: scoutQueryKeys.emissionReports(projectId, run.run_id),
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/ui/src/features/scouts/hooks/useScoutFindings.ts]: Fixed in 2f5595c88 — the report-link query key now includes `emitted_count` too, symmetric with the emissions key, so a run that emits more refetches its links and the new findings get their report chip without a manual retry/focus. (Linkage that lands *without* a count change — a finding grouping into a report minutes later — remains best-effort via staleTime, matching the per-scout section; the chip is non-essential adornment.)

File: packages/core/src/scouts/scoutFindings.ts [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+): ScoutFindingRow[] {
+  const runsById = new Map(runs.map((run) => [run.run_id, run]));
+  return emissions
+    .map((emission): ScoutFindingRow | null => {
+      const run = runsById.get(emission.run_id);
+      return run
+        ? {
+            emission,
+            run,
+            report: reportBySourceId.get(emission.source_id) ?? null,
```
[the remark this answers is not in the document]
Comment (andrewm4894) [reply packages/core/src/scouts/scoutFindings.ts:102]: This one I don't think holds, given the `source_id` scheme. The backend builds it as `run:<run_id>:finding:<finding_id>` (`products/signals/backend/models.py:1142`), so it's scoped per run — two findings from *different* runs in the flattened set can never share a `source_id`, and the link endpoint returns links with that same run-scoped `source_id`. So `reportsBySourceId` can't apply one run's report to another run's finding.

The only collision is a single run re-emitting the same `finding_id` twice → same `source_id` *and* same finding, which correctly maps to the same report (the React list is keyed by `emission.id` to keep both rows distinct). This also mirrors the existing per-scout `ScoutSignalsSection`/`ScoutEmissionCard` join, which keys by `source_id` the same way. Leaving as-is — happy to reconsider if the `source_id` format changes.
